lib/event/emitter.js

```diff
@@ -39,7 +39,12 @@ EventEmitter.prototype.emit = function(type) {
   if (!listeners) {
     return;
   }
-  var args = Array.prototype.slice.call(arguments, 1);
+  // We need to copy arguments to local array for using it in another function call.
+  // The only way js engine (like V8) will be able to optimize it.
+  var args = [];
+  for (var ai = 1; ai < arguments.length; ++ai) {
+    args.push(arguments[ai]);
```

Comment:
You're also introducing x amount function calls here. So this makes me wonder if you have even benchmarked the changes you've made.